src/module.c

```diff
@@ -1363,27 +1400,54 @@ NAPI_METHOD(dcn_start_threads) {
   dcn_context->loop_thread = 1;
   uv_thread_create(&dcn_context->imap_thread, imap_thread_func, dcn_context);
   uv_thread_create(&dcn_context->smtp_thread, smtp_thread_func, dcn_context);
+  uv_thread_create(&dcn_context->mvbox_thread, mvbox_thread_func, dcn_context);
 
   NAPI_RETURN_UNDEFINED();
 }
 
+/*
+void wait_until_idle(dc_context_t* context) {
+  // I know is_idle doesn't exist and maybe we can't do it this way, just
+  // wanted to illustrate what I'm after, also sleep(1) is maybe not ideal
+  while (!context->imap->is_idle) sleep(1);
+  while (!context->mvbox->is_idle) sleep(1);
+  while (!context->smtp->is_idle) sleep(1);
+}
+*/
```

Comment:
@r10s This is what I'd like to do before interrupting the threads. We would probably need some mutexes etc, but you get the idea at least.